### ISSUE TITLE
Crash when selecting ROI 

### ISSUE THREAD

**Issue by jxchen01** [InsightSoftwareConsortium/itk-jupyter-widgets#120]
I just encounted a very strange issue on Windows 10 laptop. 

When I do `view(IMG)`, and I select a smaller ROI on the image, I always got an error saying ""python backend is dead", then the jupyter notebook kernel crashed and restarted. I can reproduce the same problem on two different windows 10 laptops, one surface and one thinkpad t440p. 

However, everthing is fine on Windows 10 desktop. 

I guess this may be a memory issue? or graphic card issue?

Any thoughts?

BTW, I have to say this viewer is really really awesome!

Thanks,
Jianxu

**Comment by thewtex** [comment]
Hi Jianxu,

> BTW, I have to say this viewer is really really awesome!

Thanks for the kind words! They are motivating :-)

> I guess this may be a memory issue? or graphic card issue?

There are a few ways to help debug

1) Check in the terminal where Jupyter was started for any error messages.
2) Press "Ctrl + Shift + I" to bring up your browser's debugger. Check in the Console tab for any error messages.

What browser are you using?

**Comment by jchen16** [comment]
I also see this message in browser's debugger:

Active resource loading counts reached a per-frame limit while the tab was in background. Network requests will be delayed until a previous loading finishes, or the tab is brought to the foreground. See https://www.chromestatus.com/feature/5527160148197376 for more details

I am not sure whether that is related.

**Comment by thewtex** [comment]
@jchen16 please try upgrading to the recently released version, 0.15.2,

```
python -m pip install --upgrade itkwidgets
```

Does the behavior still exist?

Also, do you observe the behavior with [the repository example](https://mybinder.org/v2/gh/InsightSoftwareConsortium/itk-jupyter-widgets/master?filepath=examples/SelectRegionOfInterest.ipynb)?

**Comment by jxchen01** [comment]
Thanks for your suggestions. 

(1) update did not solve the problem

(2) running the repository example is fine on my tested laptops

So, this could be a backend thing rather than a brower issue?

**Comment by thewtex** [comment]
> So, this could be a backend thing rather than a brower issue?

Possibly. What browser is used in all cases, i.e. Chrome, Firefox, ...? Is the Jupyter Python kernel running on the local system in these cases?

**Comment by jxchen01** [comment]
I am using Chrome on tested Windows laptop and Safari on tested MacBook Pro. The python kernel is running on the local system. The kernel runs without any issue until selecting ROI.

**Comment by thewtex** [comment]
Thanks for the information.

Does the kernel crash only with specific datasets?

**Comment by jxchen01** [comment]
What I tested was images as 3D numpy array (not ITK images). I tested on arrays of large and small sizes, always having the same crash.

**Comment by thewtex** [comment]
Interesting.

Could you please share a minimal notebook that reproduces the behavior?

**Comment by jxchen01** [comment]
I will prepare a sharable noteboook to reproduce the results. 

Meanwhile, I think I find more clues.  First of all, when I say selecting ROI, I only tested under z-plane view. Today, I did some stress test. Under z-plane view, when I scoll up and down to zoom-in and zoom-out, I try to zoom-in as much as I can. I keep zooming in. At a certain point, the same error occurs. Namely, the python kernel died. This error in stress test occurs also on Desktop machine.

**Comment by jxchen01** [comment]
Hi @thewtex , I had more updates on this. 

It is hard for me have a notebook to reproduce the error. But, I tried to test the older versions one by one from the latest. I cannot remember why I did such test. But this test turned out to be very helpful. The 0.14.0 release version is the last version I can run without any aforementioned errors on all OS. Some major changes, especially related to ROI selection, from 0.14.0 to 0.15.0 caused the problem.
